### ISSUE TITLE
depends on libwebkitgtk-1.0-0 which is deprecated

### ISSUE THREAD

**Issue by teusbenschop** [postiffm/bibledit-desktop#37]
A bug was reported on Debian.

https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=866632

Package: bibledit-gtk
Severity: serious
Tags: sid buster
User: pkg-webkit-maintainers@lists.alioth.debian.org
Usertags: oldlibs libwebkitgtk-1.0-0 webkit1

Hi,

bibledit-gtk depends on libwebkitgtk-1.0-0, which is deprecated in
favor of libwebkit2gtk-4.0-37 (provided in Debian by webkit2gtk).
bibledit-gtk should be ported to the new webkitgtk version so we can
remove the old, unmaintained one.

Please try to do soon as we're going to try to remove the old webkitgtk soon.

I am filing this bug as "serious" since the old webkitgtk will not be
available in the next major stable release of Debian (codenamed
"buster").

If you have any question don't hesitate to ask.

On behalf of the Debian WebKit Maintainers,
Jeremy Bicha

If there's a new upstream version that no longer depends on libwebkitgtk-1.0, may you please update this issue at github or this bug at debian, so the package maintainers may be able to package the updated software for Debian.

**Comment by LAfricain** [comment]
> so the package maintainers may be able to package the updated software for Debian.

My dream :)  :+1:

**Comment by postiffm** [comment]
See the webkit2 branch. 

This is going to present some difficulties because of the new separate process model in webkit2. Many aspects of migration have been easy. But there are a couple--DOM handling and copy/paste/undo editing functions that cannot be done nearly as easily as before because asynchronous cross-process communication is needed. These changes are a work in progress.

**Comment by postiffm** [comment]
We have had internal discussions about ditching webkit entirely in favor of using something else. Gtkhtml? Or does the gtktextview have some built-in support we could use.

The main thing we do is generate text with links that are clickable, which then induces the program to generate another page of stuff. We need to be able to go back to the prior page...

It seems like most webkit2 support is present and compiles. I'm still OK to use it if stuff like copy/paste can be finished. It might be easiest if someone could finish this...then we could have full webkit2 support and move forward from there. This would let us do things like show simple web pages of other Bible resources right inside a window of Bibledit-Desktop.

**Comment by teusbenschop** [comment]
The urgency is no longer there in Debian.
The package is no longer in a development branch in Debian.
See the link for more information: https://tracker.debian.org/pkg/bibledit-gtk

**Comment by LAfricain** [comment]
It's pity to lose bibledit-gtk in the debian packages.
I think it's difficult to add new software on the debian repos, need advocate, supporters, etc...
And even if it's no more in a dev branch, we hope bb-desktop can runs well on debian, for this reason, the libwebkit2gtk migration is expected.

**Comment by teusbenschop** [comment]
> I don't understand. Please clarify.

Bibledit-Desktop could no longer be maintained in Debian, because Debian has dropped WebKit (version 1). That made it urgent to have Bibledit-Desktop drop WebKit (version 1) too. Later Bibledit-Desktop was dropped from Debian development. Now, since it's dropped already, the urgency is less now. As @LAfricain says, it's a pity to lose bibledit-gtk in Debian. But this has already happened. But as long as Bibledit-Desktop keeps using WebKit (version 1), it cannot be built on newer Debian versions. So here's the urgency: If someone needs Bibledit-Desktop on Debian, then it's urgent.

**Comment by LAfricain** [comment]
Please work on this. On Ubuntu 20.04 `libwebkitgtk-dev` is also removed from the repositories. Bibledit-Desktop will not work on it.

**Comment by postiffm** [comment]
Hello! Just a few days ago, I merged the gtk3 and webkit2 updates into the master branch. So...we no longer require gtk2 or libwebkitgtk-1. You should be able to rebuild. Let me know what you find.

**Comment by LAfricain** [comment]
This is great! Let me test!

**Comment by LAfricain** [comment]
Please can you update the linux/buildenvLinux.sh with the new dependancies? I don't know the exact name of it.

**Comment by postiffm** [comment]
I thought I did already. linux/buildendLinux.sh is the right file.

**Comment by LAfricain** [comment]
line 33: `apt-get install libwebkitgtk-dev` is this still relevant?

**Comment by postiffm** [comment]
Pull the latest cb56759

**Comment by LAfricain** [comment]
Compiled! beautiful! Probably we need to add libgtksourceview-3.0-dev and libwebkitgtk-3.0-dev, in the buildenvlinux.sh file.
Now I will  test it, and try to push it on my ppa.
I just noticed a issue with the checks for chapter and verse. When I check, and click on the url with an error, i have this message: `The URL can’t be shown`.
I can open a new issue, this one can be closed

**Comment by postiffm** [comment]
Did it fail to build without libwebkitgtk-3.0-dev? My memory was that since we moved to webkit2, libwebkitgtk-3.0 was not relevant now.

**Comment by LAfricain** [comment]
You are correct, libwebkitgtk-3.0 is enough. This issue can be closed ;)

**Comment by postiffm** [comment]
@LAfricain The only thing that is not done here is that the package is not back in the debian distribution channel. Is that something you can help with?

**Comment by LAfricain** [comment]
I don't know... Maybe @teusbenschop can help us?

**Comment by teusbenschop** [comment]
I can probably help you in some way. Or my colleagues Debian Developers can do it, if they get asked.

If you @LAfricain provide the Debian packaging, one of the Debian Developers at the Crosswire Packaging Team (https://qa.debian.org/developer.php?email=pkg-crosswire-devel%40alioth-lists.debian.net) probably could submit it for inclusion.

It is a process that requires multiple steps:

* Create the Debian packaging.
* Build the Debian package, and test it.
* Have the package reviewed by Debian Developer(s) / Mentor(s).
* Get the copyright licenses right in the package.
* Submit it to the Debian archives.
* Have it reviewed by the Debian FTP Masters.
* Get it into the Debian Archives.

**Comment by LAfricain** [comment]
> I can probably help you in some way. Or my colleagues Debian Developers can do it, if they get asked.

This is great! thank you.
About me, Alas I really don't have the skills to make a debian package. But I still tried on my ppa, however there is an error during the compilation on launchpad. You can take a look at it Teus: [buildlog](https://launchpadlibrarian.net/480938165/buildlog_ubuntu-bionic-amd64.bibledit-desktop_4.17-ubuntu18.04.4_BUILDING.txt.gz)
I tweak more than anything else just because I am very motivated. Although I am willing to learn. I know that with bibledit-cloud the ppa and surely also with debian the process is automatic as soon as there are new versions. Is this correct? And in addition the translation can be done on launchpad. Which is very interesting.

But the debian package is almost ready. It's only on launchpad where the build failed, if I use the command:
```
debuild -us -uc -ui -I
```
The package is well build.

**Comment by teusbenschop** [comment]
> I know that with bibledit-cloud the ppa and surely also with debian the process is automatic as soon as there are new versions. Is this correct? 

If the developer has made some scripts to automate stuff, then a new version is easy to get into the PPA  of Ubuntu, and into Debian.

> But the debian package is almost ready. It's only on launchpad where the build failed, if I use the command:

Once the Debian package is ready, I am willing to review it, and provide feedback for you on the package.

**Comment by LAfricain** [comment]
> Once the Debian package is ready, I am willing to review it, and provide feedback for you on the package.

It's for ubuntu, but it will  probably work for debian. What do you need? You can download everything you need on my ppa or on this link: https://send.firefox.com/download/c9049044bddacaa6/#ECjL1tmYMCn64VlVVwfraw

**Comment by teusbenschop** [comment]
It helps that it is there already for Ubuntu.

What I would need is a package that 
1. has been tested to build on Debian.
2. has been tested to work there without errors.

More information on creating a Debian package: https://wiki.debian.org/Packaging

**Comment by LAfricain** [comment]
> has been tested to build on Debian.

OK I need to use a VM because I'm an Ubuntu user, not Debian. Is Debian 10.4 a good version?

**Comment by teusbenschop** [comment]
> OK I need to use a VM because I'm an Ubuntu user, not Debian. Is Debian 10.4 a good version?

For packaging you will need to run Debian Unstable.

https://wiki.debian.org/DebianUnstable

That is also called "Debian Sid". It is the newest version of Debian, it has rolling updates, and always contains the newest packages.

**Comment by LAfricain** [comment]
Understood! Then 10.4 stable is ok, but I just change the sources.list.

**Comment by LAfricain** [comment]
- Debian installation OK
- Ubuntu BBD package installed on Debian unstable OK, it works

-  Now preparation of an unstable debian package: work in progress

**Comment by teusbenschop** [comment]
It's going good, good luck with creating a package for Debian Unstable, bro!

**Comment by LAfricain** [comment]
Please help, I can't find the package `libwebkitgtk-3.0` for Ubuntu 20.04, but it seems to be present in debian unstable? Any idea? Did the package change its name? Is the package mandatory for BDD?

**Comment by teusbenschop** [comment]
The package ```libwebkitgtk-3.0``` is not available in Debian Unstable either.

A search for "libwebkitgtk" in Debian gives no results.
A search for "libebkit2gtk" in Debian gives some results:

```
$ apt search libwebkit2gtk
Sorting... Done
Full Text Search... Done
libwebkit2gtk-4.0-37/unstable,now 2.28.2-1 amd64 [installed,automatic]
  Web content engine library for GTK

libwebkit2gtk-4.0-37-gtk2/unstable 2.28.2-1 all
  Transitional dummy package

libwebkit2gtk-4.0-dev/unstable,now 2.28.2-1 amd64 [installed]
  Web content engine library for GTK - development files

libwebkit2gtk-4.0-doc/unstable 2.28.2-1 all
  Web content engine library for GTK - documentation
```

I cannot find anything in Debian unstable that looks like "libgtkhtml":

```
$ apt search libgtkhtml
Sorting... Done
Full Text Search... Done
```

**Comment by teusbenschop** [comment]
For the Bibledit-Desktop software to be able to be built on the newest Debian and Ubuntu versions, it should not require libwebkitgtk-3.0 and libgtkhtml3.14-dev.

**Comment by LAfricain** [comment]
Ok I try without this package. Thank you. I'll change the Linux dep in the script.

**Comment by LAfricain** [comment]
Build failed for ubuntu 18.04 and I don't know why:
https://launchpadlibrarian.net/481196533/buildlog_ubuntu-bionic-amd64.bibledit-desktop_4.17-ubuntu18.04.5_BUILDING.txt.gz

**Comment by teusbenschop** [comment]
The reason that the build failed usually can be found towards the end of the build log.

Starting at the end of the build log, you may gradually scroll up the build log, till the errors become visible in the build log.

In your case, it's this fragment:

```
configure:20937: checking for GTKSOURCEVIEW
configure:20944: $PKG_CONFIG --exists --print-errors "gtksourceview-3.0 >= 3.0.0"
Package gtksourceview-3.0 was not found in the pkg-config search path.
Perhaps you should add the directory containing `gtksourceview-3.0.pc'
to the PKG_CONFIG_PATH environment variable
No package 'gtksourceview-3.0' found
configure:20947: $? = 1
configure:20961: $PKG_CONFIG --exists --print-errors "gtksourceview-3.0 >= 3.0.0"
Package gtksourceview-3.0 was not found in the pkg-config search path.
Perhaps you should add the directory containing `gtksourceview-3.0.pc'
to the PKG_CONFIG_PATH environment variable
No package 'gtksourceview-3.0' found
configure:20964: $? = 1
configure:20978: result: no
No package 'gtksourceview-3.0' found
configure:20994: error: Package requirements (gtksourceview-3.0 >= 3.0.0) were not met:

No package 'gtksourceview-3.0' found

Consider adjusting the PKG_CONFIG_PATH environment variable if you
installed software in a non-standard prefix.

Alternatively, you may set the environment variables GTKSOURCEVIEW_CFLAGS
and GTKSOURCEVIEW_LIBS to avoid the need to call pkg-config.
See the pkg-config man page for more details.
```

**Comment by LAfricain** [comment]
Success!! Peter the package was build successfully on debian unstable. How can I send you? Send firefox is ok for you?

**Comment by teusbenschop** [comment]
Great!
Yes Firefox send is okay

**Comment by LAfricain** [comment]
All is [here](https://send.firefox.com/download/79993409eac0ee61/#truHgGe5TsAVljrZA9YxHg).

**Comment by postiffm** [comment]
> 
> 
> Ok I try without this package. Thank you. I'll change the Linux dep in the script.

Did you remove these packages? Did the build work without them?

**Comment by LAfricain** [comment]
> Did you remove these packages?

No

> Did the build work without them?
Yes very well. And all the build problem were related to this dep. I just update the linux scripti in #132

**Comment by LAfricain** [comment]
You can update your documentation for Ubuntu. I was just accepted in the Crosswire ppa, and I push the new release of bibledit-Desktop on it (https://launchpad.net/~pkgcrosswire/+archive/ubuntu/ppa). Even if with the problem of the checks it's probably better to use the 4.16 version, present  on my ppa.

**Comment by postiffm** [comment]
Checks are now fixed.

**Comment by LAfricain** [comment]
Updated in the crosswire ppa. the closing windows (tidying up) shows `_Cancel` rather `Cancel`.
Line 112 should be:
`  GtkWidget * button = gtk_button_new_with_label (_("Cancel"));`

**Comment by teusbenschop** [comment]
> All is [here](https://send.firefox.com/download/79993409eac0ee61/#truHgGe5TsAVljrZA9YxHg).

Thank you for that. I am intending to start looking at this package.

**Comment by teusbenschop** [comment]
Here is a link to the ITP (Intent To Package) bug report on the Debian bug tracking system.

https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=961535

This is one of the initial steps to announce the intention to package this Bibledit-desktop software for Debian.

**Comment by postiffm** [comment]
Cancel button fixed in 569b60c

**Comment by LAfricain** [comment]
@teusbenschop Now how can I do if I want to propose a new package,is it better to wait the approval for the first, or is it better to propose a new one directly?

**Comment by teusbenschop** [comment]
I am not sure what to answer to this because it is not entirely clear to me what this is about. 

What is the "new" package, and what is the "first" package in your case?

**Comment by LAfricain** [comment]
:) The first is the package I send and prepare for you, but during this days Matthew solved a important bug according me, and I already update the BBD package on the crosswire ppa for Ubuntu.
Then is it better to continue the process with the current package you have, or to build a new one with the last commit?

**Comment by teusbenschop** [comment]
Okay, now I understand what this is about.

The Debian package still has to be started. So it is good to send me the newest code you have, with the fix in it.

Another issue is that building a Debian package needs the tarball. If you could send me the newest tarball, that would be the most helpful.

**Comment by teusbenschop** [comment]
There are the official releases of Bibledit-Desktop at https://github.com/postiffm/bibledit-desktop/releases.

I prefer to use one of the official releases for creating a Debian package from.

Is there any problem with using the newest official release?

**Comment by LAfricain** [comment]
The deb package was build with the 4.17 release. I have no idea if the current 4.17 is already uptodate with the last commits I guess no.
Now, I'm lost, what do you need :(

**Comment by teusbenschop** [comment]
After that I found the Releases on GitHub, I downloaded version 4.17 of Bibledit-desktop. I think this is all that I need for just now. (Plus the Debian packaging you already sent me). Nothing more is needed now.

**Comment by LAfricain** [comment]
:+1:

**Comment by teusbenschop** [comment]
The new proposed packaging is at https://salsa.debian.org/debian/bibledit-desktop. 

But the packaging is not yet complete. It lacks a complete debian/copyright file.

The debian/copyright file cannot yet be created because lots of copyright information is lacking.

At Debian they are particular about the copyright. More information is at https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/.

Running the command 
```licensecheck --check '.*' --recursive --deb-machine --lines 0 *``` 
gives lots of information about lacking copyright information.

I ran the command for you and am attaching the resulting file here.

[licensecheck.txt](https://github.com/postiffm/bibledit-desktop/files/4702151/licensecheck.txt)

Question: May you provide all license information that is still needed.

After all license information is there, I can proceed with the packaging and submit it to Debian for their review.

**Comment by postiffm** [comment]
I will take a look at this. I don't believe I am quite sure that I will NOT be able to add copyright headers to all those files. Some are datafiles that might better reside in another repo. I'll have to check into it and understand more the requirements.

**Comment by teusbenschop** [comment]
If it may help you to take a look at the Bibledit-Cloud package that was approved for inclusion in Debian not too long ago.

https://salsa.debian.org/debian/bibledit-cloud

There's data files, as you mentioned.
For example in Bibledit-Cloud there's a text file:
https://salsa.debian.org/debian/bibledit-cloud/-/blob/master/versification/Original.txt
The copyright header was not added to the text file itself as that was not convenient.

But the copyright for those files was stated as follows:

```
Files: *
 debian/*
Copyright: 2003-2019 Teus Benschop
License: GPL-3.0+
```

The Bibledit-Desktop package contains material from Bibles International.
If the copyright was stated somehow as above, that would be sufficient.

One important requirement would be that the copyright is free and is compatible with Debian's view of what "free" is.

Anyway, thanks for going to look into this, and don't hesitate to ask questions. Good luck!
